FlowCrypt/src/main/java/com/flowcrypt/email/util/OutgoingMessagesManager.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
+ * Contributors: denbond7
+ */
+
+package com.flowcrypt.email.util
+
+import android.content.Context
+import com.flowcrypt.email.database.FlowCryptRoomDatabase
+import com.flowcrypt.email.database.entity.MessageEntity
+import com.flowcrypt.email.extensions.java.lang.printStackTraceIfDebugOnly
+import com.flowcrypt.email.security.KeyStoreCryptoManager
+import jakarta.mail.Message
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.File
+
+/**
+ * @author Denys Bondarenko
+ */
+object OutgoingMessagesManager {
+  private const val DIRECTORY_OUTGOING = "outgoing"
+
+  fun getOutgoingMessagesDirectory(context: Context): File {
+    return FileAndDirectoryUtils.getDir(DIRECTORY_OUTGOING, context.filesDir)
+  }
+
+  /**
+   * Store [Message] in [DIRECTORY_OUTGOING] folder.
+   * This folder is located in the 'files' folder in the app private root directory.
+   * Need to add that [Message] will be encrypted by Android KeyStore via [KeyStoreCryptoManager]
+   * and the file will use [MessageEntity.id] as a name.
+   *
+   * @param context              Interface to global information about an application environment.
+   * @param id                   This value will be used as a file name.
+   * @param mimeMessage          the outgoing MIME message
+   */
+  suspend fun enqueueOutgoingMessage(
+    context: Context,
+    id: Long,
+    mimeMessage: Message
+  ) = withContext(Dispatchers.IO) {
+    val directory = getOutgoingMessagesDirectory(context)
+    val file = File(directory, "$id")
+    file.outputStream().use { outputStream ->
+      KeyStoreCryptoManager.encryptOutputStream(outputStream) { cipherOutputStream ->
+        cipherOutputStream.use { mimeMessage.writeTo(it) }
+      }
+    }
+  }
+
+  /**
+   * Update existing [Message] in [DIRECTORY_OUTGOING] folder.
+   * [Message] will be encrypted by Android KeyStore via [KeyStoreCryptoManager]
+   * and the file will use [MessageEntity.id] as a name.
+   *
+   * @param context              Interface to global information about an application environment.
+   * @param id                   This value will be used as a file name.
+   * @param mimeMessage          the outgoing MIME message
+   */
+  suspend fun updatedOutgoingMessage(
```

Comment:
probably it should be `updateOutgoingMessage` instead of `updatedOutgoingMessage`?